hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/transfer/ClassicTransfersDecoder.java

```diff
@@ -328,15 +330,33 @@ private TokenTransferList adjustingUnits(
                 .build();
     }
 
+    /**
+     * In a case of transferring funds from one account to another, ideally the ordering of (debit, account1)
+     * and (credit, account2), or vice versa, would not produce different results. However, in order to match
+     * mono service behavior, the ordering must be specified in certain cases. Therefore, we define the
+     * following enum to indicate which of the debit or credit should be added first to the transfer list.
+     */
+    private enum TransferPrecedence {
+        // A value of DEBIT indicates that the debit account amount should be added first
+        DEBIT,
+        CREDIT
+    }
+
     private TokenTransferList sendingUnitsFromTo(
             @NonNull final TokenID tokenId,
             @NonNull final AccountID from,
             @NonNull final AccountID to,
             final long amount,
-            final IsApproval isApproval) {
+            final IsApproval isApproval,
+            @NonNull final TransferPrecedence precedence) {
         final var accountAmounts = new ArrayList<AccountAmount>();
-        accountAmounts.add(debit(from, amount, isApproval));
-        accountAmounts.add(credit(to, amount));
+        if (precedence == TransferPrecedence.DEBIT) {
```

Comment:
yikes.  this is scary that this matters.

Reply:
The only difference it made in the record stream was a different ordering of account amounts. This is purely to match mono service behavior; we can probably remove this pretty quickly after deployment (and I suspect we probably should, cuz agreed – this shouldn't matter)